exercises/src/main/scala/fpinscala/laziness/Stream.scala

```diff
@@ -17,20 +27,95 @@ trait Stream[+A] {
     case Empty => None
     case Cons(h, t) => if (f(h())) Some(h()) else t().find(f)
   }
-  def take(n: Int): Stream[A] = ???
 
-  def drop(n: Int): Stream[A] = ???
+  //5.2
+  def take(n: Int): Stream[A] = this match {
+    case _ if n == 0 => empty
+    case Cons(h, _) if n == 1 => cons(h(), empty)
+    case Cons(h, t) => cons(h(), t().take(n - 1))
+  }
+
+  //5.2
+  def drop(n: Int): Stream[A] = this match {
+    case Cons(_, t) if n >= 1 => t().drop(n - 1)
+    case _ => this
```

Comment:
`_` means `Empty`, right? Probably clearer to use `Empty` instead. Doesn't really apply here, but also better practice since it's more resilient to having other cases added.